src/librustc_driver/pretty.rs

```diff
@@ -255,7 +255,9 @@ trait HirPrinterSupport<'hir>: pprust_hir::PpAnn {
 
     /// Computes an user-readable representation of a path, if possible.
     fn node_path(&self, id: ast::NodeId) -> Option<String> {
-        self.hir_map().and_then(|map| map.def_path_from_id(id)).map(|path| {
```

Comment:
This uses `def_path_from_id` which is not equivalent to `def_path_from_hir_id`.

Reply:
Then I was mistaken. Do you have a suggestion?

Reply:
`def_path_from_hir_id` is unused. I would just change it to do the same thing as `def_path_from_id`.

Reply:
I think I don't follow. Both functions essentially call `opt_local_def_id` modulo a conversion from `HirId`.

Reply:
`def_path_from_hir_id` panics if there no corresponding `DefPath` while `def_path_from_id` returns `None` in that case.

Reply:
Yeah, I think this was just a copy-paste mishap when I created that one; it should be adjusted so that it is equivalent to `def_path_from_id`.

Reply:
Ah, I see! Sure, I will change it.